portal-bridge/src/bridge/history.rs

```diff
@@ -147,13 +157,15 @@ impl HistoryBridge {
             // from block_index to latest_block.
             for height in block_index..=latest_block {
                 self.metrics.report_current_block(height as i64);
+                let permit = self.acquire_gossip_permit().await;
                 Self::spawn_serve_full_block(
                     height,
                     None,
                     self.portal_client.clone(),
                     self.execution_api.clone(),
-                    None,
+                    permit,
                     self.metrics.clone(),
+                    self.census_tx.clone(),
                 );
             }
```

Comment:
Why was a semaphore added to latest mode? I get why we would do it for backfill, but if we are only offering to 8 Enr's every 12 seconds we shouldn't have a issue.

If anything I would make it if we reach the gossip limit the code panics, as if we can't keep up with latest something is wrong in a big way and we need to improve the code to be able to keep up

Reply:
I added a semaphore primarily for consistency across the bridges. I didn't see any strong reason to **not** include a semaphore here.
- It's fairly lightweight code, and if it's never actually used to rate limit, then that's fine
- Sooner or later it would be nice to refactor this bridge entirely, so improving the consistency across bridge implementations is a step towards this direction
- Pegging the estimate here against 8 enrs / second is not ideal (as mentioned in the other comment, I'm not a huge fan of just picking 8 as the permanent limit)
  - Even if we choose 8, it would be 24 offers / 12 seconds, taking into account the various content types, but that doesn't really affect the point you're making
- It's possible that we gossip more than a block / 5 second loop. eg, maybe there's a bad http response from the provider causing us to miss a few and then process multiple blocks in a later loop 
  - If we update to a webhook provider api to get latest blocks, this would be avoided, but pandaops nodes don't support webhooks right now

Reply:
> * Sooner or later it would be nice to refactor this bridge entirely, so improving the consistency across bridge implementations is a step towards this direction

In that case I think the state bridge should be it's own implementation. Our limiting factor is the State network and I think all design decisions should be made with State in mind first as it is the hardest problem, and if we can get state to work, then the other networks will work too.

> * Pegging the estimate here against 8 enrs / second is not ideal (as mentioned in the other comment, I'm not a huge fan of just picking 8 as the permanent limit)

If we can only do X offer's a second but we have Y State content to gossip in Z seconds. If the data is successfully gossiped to a node it can handle neighborhood gossip from their, our priority is being able to gossip a latest diff in ideally less then 8 seconds. (we should aim for way less then 8 seconds)

If we gossip to even 1 enr that is enough as long as the bridge is able to gossip the whole diff or the whole diff split across multiple bridges in under the time requirement

>   * Even if we choose 8, it would be 24 offers / 12 seconds, taking into account the various content types, but that doesn't really affect the point you're making

State is at the end of the day the hardest problem, it is also the higher priority out of all the network's. If we can handle state we can more then handle the rest

> * It's possible that we gossip more than a block / 5 second loop. eg, maybe there's a bad http response from the provider causing us to miss a few and then process multiple blocks in a later loop

We should stop relying on the JSON-RPC as it is unreliable. ^ The reasons why we should do this is outlined in the quote above
  
>   * If we update to a webhook provider api to get latest blocks, this would be avoided, but pandaops nodes don't support webhooks right now

Even then it is unreliable, to a certain degree it makes sense for the State bridge to also handle gossiping History, and Beacon as it will get the data faster, as it will be directly executing and receiving the block from the CL client itself.